### PR TITLE
feat: add support for tedious v17

### PR DESCRIPTION
**Pull request by david-luna**
v17 release does not have major changes but just drops node versions support.

### Checklist

- [ ] Implement code
- [ ] Add tests
- [ ] Update TypeScript typings
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: .tav.yml [hunk 1]
```diff
@@ -335,6 +335,9 @@ tedious:
   - versions: '16.0.0 || 16.1.0 || >16.1.0 <17' # first and last majors subset of '16.x'
     node: '>=16'
     commands: node test/instrumentation/modules/tedious.test.js
+  - versions: '>=17.0.0 <18' # first and last majors subset of '17.x' (as for now there is only v17.0.0)
+    node: '>=20'
```
Comment (trentm) [inline .tav.yml:339]:
```suggestion
    node: '18.x || >=20'
```

IIUC tedious@17 still supports Node v18: https://github.com/tediousjs/tedious/compare/v16.7.1...v17.0.0#diff-e98936aa52a6dd7416e4296e9628456227d834f7245967383fd9ff80fd985dadR49

Comment (david-luna) [reply .tav.yml:339]: yup. I misunderstood the commit message

**Review by trentm — CHANGES_REQUESTED** [review]
I think also want this change:

```diff
diff --git a/test/instrumentation/modules/tedious.test.js b/test/instrumentation/modules/tedious.test.js
index 085e0dca..5a342339 100644
--- a/test/instrumentation/modules/tedious.test.js
+++ b/test/instrumentation/modules/tedious.test.js
@@ -24,6 +24,7 @@ const tediousVer =
   require('../../../node_modules/tedious/package.json').version;
 const semver = require('semver');
 if (
+  (semver.gte(tediousVer, '17.0.0') && semver.lt(process.version, '18.0.0')) ||
   (semver.gte(tediousVer, '16.0.0') && semver.lt(process.version, '16.0.0')) ||
   (semver.gte(tediousVer, '15.0.0') && semver.lt(process.version, '14.0.0')) ||
   (semver.gte(tediousVer, '12.0.0') && semver.lt(process.version, '12.3.0')) ||
```

or similar. However, tedious@17.0.0 hasn't yet made any changes that actually break on node v16.

**Comment by trentm** [comment]
@david-luna Did you see this comment? https://github.com/elastic/apm-agent-nodejs/pull/3901#pullrequestreview-1917915966

**Comment by david-luna** [comment]
> @david-luna Did you see this comment? [#3901 (review)](https://github.com/elastic/apm-agent-nodejs/pull/3901#pullrequestreview-1917915966)

sorry no, I was distracted the the mongodb issue in OTel. Your propose change makes sense, I'm going to apply it now.
Thanks!

**Review by trentm — APPROVED** [review]

**Pull request merged**